detection_rules/index_mappings.py

```diff
@@ -173,12 +173,16 @@ def prune_mappings_of_unsupported_types(
         delete_nested_key_from_dict(stream_mappings, field_name)
     nested_flattened_fields = find_flattened_fields_with_subfields(stream_mappings)
     for field in nested_flattened_fields:
+        # Remove both .fields and .properties entries for flattened fields
+        # .properties entries can occur when being merged with non-ecs or custom schemas
         field_name = str(field).split(".fields.")[0].replace(".", ".properties.") + ".fields"
+        property_name = str(field).split(".fields.")[0].replace(".", ".properties.") + ".properties"
```

Comment:
Takes a field path like `azure.signinlogs.properties.authentication_details`. The logic will 
- Replaces all dots with `.properties.` to get: `azure.properties.signinlogs.properties.properties.properties.authentication.properties.details`
- Adds the suffix (`.fields` or `.properties`) to get the final path

At a glance this seems incorrect, but I think later we call `delete_nested_key_from_dict` to correctly delete the correct keys. Is that right?

Does it make more sense to insert `.properties.` only between field name segments, vs just replacing all dots for code readability?

Reply:
Updated.